src/pages/docs/chat/api/javascript/messages.mdx

```diff
@@ -0,0 +1,371 @@
+
+`OperationMetadata: Record<string, unknown>`
+
+### Returns
+
+Returns a promise. On success, the promise is fulfilled with the updated [`Message`](/docs/chat/api/javascript/message) object. On failure, the promise is rejected with an [`ErrorInfo`](/docs/chat/api/javascript/error-info).
+
+<Code>
+```javascript
+const updatedMessage = await room.messages.update(
+  message.serial,
+  {
+    text: 'Updated message text',
+    metadata: { edited: true, editedAt: Date.now() }
+  },
+  {
+    description: 'Fixed typo'
+  }
+);
+```
+</Code>
+
+---
+
+## Delete a message <a id="delete"/>
+
+`room.messages.delete()`
+
+Delete a message in the chat room.
+
+This method performs a 'soft' delete, meaning the message is marked as deleted.
+
+Note that the Promise may resolve before or after the message is deleted from the realtime channel. This means you may see the message that was just deleted in a callback to `subscribe` before the returned promise resolves.
+
+The [`Message`](/docs/chat/api/javascript/message) instance returned by this method is the state of the message as a result of the delete operation. If you have a subscription to message events via `subscribe()`, you should discard the message instance returned by this method and use the event payloads from the subscription instead.
+
+Should you wish to restore a deleted message, and providing you have the appropriate permissions, you can simply send an update to the original message.
+
+`delete(serial: Serial, deleteMessageParams?: DeleteMessageParams): Promise<Message>`
+
+### Parameters
+
+Use the following parameters:
+
+| Parameter | Description | Type |
+| --------- | ----------- | ---- |
+| `serial` | A string or object that conveys the serial of the message to delete. | [`Serial`](#serial) |
+| `deleteMessageParams` | *Optional* Optional details to record about the delete action. | [`DeleteMessageParams`](#deletemessageparams) |
+
+#### DeleteMessageParams
+
+The following `DeleteMessageParams` can be set:
+
+| Parameters | Description | Type |
+| ---------- | ----------- | ---- |
+| `description` | *Optional* A description of why the message was deleted. | String |
+| `metadata` | *Optional* Metadata associated with the deletion. Do not use metadata for authoritative information. There is no server-side validation. When reading the metadata, treat it like user input. | [`OperationMetadata`](#operationmetadata) |
+
+### Returns
+
+Returns a promise. On success, the promise is fulfilled with the deleted [`Message`](/docs/chat/api/javascript/message) object. On failure, the promise is rejected with an [`ErrorInfo`](/docs/chat/api/javascript/error-info).
+
+<Code>
+```javascript
+const deletedMessage = await room.messages.delete(
+  message.serial,
+  {
+    description: 'Message violated community guidelines',
+    metadata: { reason: 'inappropriate content' }
+  }
+);
+```
+</Code>
+
+---
+
+## Get message history <a id="history"/>
+
+`room.messages.history()`
```

Comment:
Two things on this one:

1. We need to describe `historyBeforeSubscribe` in this section
2. We should definitely call out that a lot of of the time, you probably want that method and not this one - and we should probably document how to choose each.

Reply:
👍🏻